test/dom.cpp

```diff
@@ -23,30 +23,108 @@
  */
 #include <migraphx/dom_info.hpp>
 #include <migraphx/program.hpp>
+#include <migraphx/ranges.hpp>
+#include <migraphx/iterator_for.hpp>
 #include <basic_ops.hpp>
 #include <test.hpp>
 
+bool dominates_self(const migraphx::dominator_info& dom, const migraphx::module& m)
```

Comment:
I don't see where this is used.

Reply:
It is used in second test. 

@pfultz2  Question for curiosity, Is it ever possible for a node to dominate itself in ML Graph ? I can't think of a case like that.

Reply:
Node cannot strictly dominate itself by [definition](https://en.wikipedia.org/wiki/Dominator_(graph_theory)):

> A node d strictly dominates a node n if d dominates n and d does not equal n.